test/Microsoft.AspNet.Mvc.Test/MvcServiceCollectionExtensionsTest.cs

```diff
@@ -79,6 +83,71 @@ public void WithControllersAsServices_ScansControllersFromSpecifiedAssemblies()
             Assert.Equal(ServiceLifetime.Singleton, services[3].Lifetime);
         }
 
+        [Fact]
+        public void MultiRegistrationServiceTypes_AreRegistered_MultipleTimes()
```

Comment:
Would like to see a comment here, this kind of test is obtuse and if you 'break' it in the future on purpose, you need to know what to do to fix it.